modules/code-deployment/src/main/java/org/apache/ignite/internal/deployunit/DeploymentUnitAccessorImpl.java

```diff
@@ -36,7 +37,7 @@ public DeploymentUnitAccessorImpl(FileDeployerService deployer) {
      * {@inheritDoc}
      */
     @Override
-    public DisposableDeploymentUnit acquire(DeploymentUnit unit) {
+    public synchronized DisposableDeploymentUnit acquire(DeploymentUnit unit) {
```

Comment:
You synchronize by accessor instance which is not the best choice here. The completely independent process that is needed `unit1` will be blocked by another process that runs `computeIfNotAcquired` for `unit2`.

Reply:
I'm not sure that is going to be a bottle neck